### PR TITLE
Add commercial license configurations guide

### LINKED ISSUE #1619: Add License configuration guide
Add a guide on how to configure the commercial license for projects that needs them. This guide should be placed under the [configuration chapter](https://vaadin.com/docs/latest/configuration), and it should describe:

- That no license is needed if the developer is only using the core, free components
- That no license is needed for production use (except for Collaboration Engine)
- How the license works (namely, by checking through vaadin.com login)
- How to use licenses with TestBench, Designer, or MPR (namely, by either placing the license file in the home directory or adding the license as a system property)
- How to use license with CI/CD server
- How to troubleshoot common license issues

These topics are already covered in https://vaadin.com/licensing-faq-and-troubleshooting, but should be added to the docs for discoverability and to remove irrelevant stuff such as those related to older Vaadin versions.

### PR DESCRIPTION
**Pull request by dmitrilc**
Fixes #1619

### CODE REVIEW

**Comment by dmitrilc** [comment]
Sections not included:

**Because of relation to older versions**

- How do I see which Vaadin version the application is using?
- How do I install Vaadin 6/7/8 license keys
- I have a perpetual license for Designer 2, but after updating Designer I’m asked for a Pro subscription?

**Duplicate info with License Validation Process**
- I have Vaadin 10-23.1 application and I have problems license problems (not Designer or TestBench, see below)

**Review by jouni — COMMENTED** [review]
Here’s the terminology I’d like to propose:

- **License**: part of your subscription, which grants you the right to use our commercial products
- **License key**: the file that is placed on your computer/server that is used to check your license

I’m a little unsure if ”machine ID” should be capitalized or not. I think I would keep it as **machine ID**, not Machine ID. I don't think it’s a large-enough concept to warrant a capital letter. A bit like “user name” vs “User Name”.

**Comment by dmitrilc** [comment]
> Here’s the terminology I’d like to propose:
> 
>     * **License**: part of your subscription, which grants you the right to use our commercial products
> 
>     * **License key**: the file that is placed on your computer/server that is used to check your license
> 
> 
> I’m a little unsure if ”machine ID” should be capitalized or not. I think I would keep it as **machine ID**, not Machine ID. I don't think it’s a large-enough concept to warrant a capital letter. A bit like “user name” vs “User Name”.

Thank you so much for the review. I don't really have a strong opinion on whether "machine id" should be capitalized or not. It was capitalized at https://vaadin.com/pro/validate-license, so I just followed the convention there.

The clear distinction between a "license" and a "license key" that you have proposed is on point. The license mostly stays the same, but the license key can be re-generated over and over again. I will scan the doc to find instances where these two terms have been misused.

File: articles/configuration/licenses.adoc [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
++
+If defined, the `vaadin.proKey` system property will be used instead of the file in the home directory.
+The format of the system property is
++
+[source]
+-Dvaadin.proKey=<login-email>/<key starting with pro->
+
+For CI/CD servers without internet access, you need to download an **offline CI license**.
+This offline license can only be used for production builds.
+You can get this license by:
```
[the remark this answers is not in the document]
Comment (dmitrilc) [reply articles/configuration/licenses.adoc:82]: @jouni Thank you for bringing this up. 

The information at the link https://vaadin.com/pro/validate-license does not mention "server license key" and the instruction there still use the term "offline license". 

Do you think we should wait until this "sever license key" procedure becomes official first before using it in this guide?

**Comment by dmitrilc** [comment]
Hi @jouni, 

Since you are also reviewing this PR, there is something else that I want to bring to your attention because I believe this could be beneficial to you. 

I see a lot of information regarding old versions of Vaadin 6,7,8 at https://vaadin.com/pro/licenses. Should we also keep this page in sync with this license guide? Or maybe just link to this guide, so we don't have to repeat ourselves? As of right now, there are three places that discuss licenses.

1. The new file from this PR.
2. https://vaadin.com/licensing-faq-and-troubleshooting
3. https://vaadin.com/pro/licenses

I don't see an "edit" button for the page at https://vaadin.com/pro/licenses, so I guess this is internal.

File: articles/configuration/licenses.adoc [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
++
+.Windows
+[source,terminal]
+%HOMEPATH%\.vaadin\proKey 
++
+.macOS and Linux
+[source,terminal]
+~/.vaadin/proKey
++
+If you do not have a license installed on your development machine, you can also download your personal [filename]#proKey# file from https://vaadin.com/pro/validate-license[vaadin.com/pro/validate-license].
```
[the remark this answers is not in the document]
Comment (jouni) [reply articles/configuration/licenses.adoc:70]: The conclusion was that, no, we don't want to promote this approach any more, and will guide users towards offline and server license keys instead.
```suggestion
```

File: articles/configuration/licenses.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (jouni) [reply articles/configuration/licenses.adoc:82]: The license validation page is under development at the moment, and should become something like this in the near future: 

Screenshot 2022-08-30 at 15 40 08

That said, we should wait until that work finishes, before merging this PR. The final URL for the license key page is could still change.

File: articles/configuration/licenses.adoc [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+-Dvaadin.proKey=<login-email>/<key starting with pro->
+
+For CI/CD servers without internet access, you need to download an **offline CI license**.
+This offline license can only be used for production builds.
+You can get this license by:
+
+. Going to https://vaadin.com/pro/validate-license.
+. Selecting *Generate and offline license for CI only*.
+
+This will generate an [filename]#offlineKey# file that you should place in the _home_ directory of the CI server:
```
Comment (jouni) [inline articles/configuration/licenses.adoc:87]:
```suggestion
You should place the [filename]#offlineKey# file in the _home_ directory of the CI server:
```

File: articles/configuration/licenses.adoc [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+[source]
+~/.vaadin/offlineKey (macOS/Linux) or %HOMEPATH%\.vaadin\offlineKey (Windows)
```
Comment (jouni) [inline articles/configuration/licenses.adoc:90]:
```suggestion
[.example]
--
[source]
----
<source-info group="Windows"></source-info>
%HOMEPATH%\.vaadin\offlineKey
----

[source]
----
<source-info group="macOS/Linux"></source-info>
~/.vaadin/offlineKey
----
--
```

File: articles/configuration/licenses.adoc [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+In some cases, a popup message will appear in the browser window indicating that the license has not been validated.
+Specifically, you might see a message that says "_This Application Is Using Components Which Are A Part Of A Vaadin Subscription_".
+
+To address this, you should first verify that you have a valid license by going to https://vaadin.com/pro/validate-license.
+Once verified, clear Vaadin cookies and try again, making sure that you use the same browser window.
+
+Sometimes, ad-blockers or 3rd party cookie denial could be blocking the license check.
+If this is the case, the error message *only* appears in development mode when in localhost.
+When the application is deployed into production and accessed from outside it will not show, thus the message *is not be visible to the end users* of the application.
```
Comment (jouni) [inline articles/configuration/licenses.adoc:112]: No longer relevant after 23.1/2 and the new license checker, which does not rely on cookies.
```suggestion
```

File: articles/configuration/licenses.adoc [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+Check if you have a valid license by going to https://vaadin.com/pro/validate-license.
+If not, either your company's license administrator has not yet assigned a seat for you, or the license is no longer valid.
+
+=== SSLHandshakeException With Multiplatform Runtime
+
+There is a known error in `SSLHandshakeException` reported by users of WebSphere Liberty and WildFly in Docker: https://vaadin.com/forum/thread/18444677/running-mpr-project-on-websphere-liberty-fails-with-suncertpathbuilderexcep.
+
+=== Where Can I Get Help With License Related Issues?
+
+You can contact license@Vaadin.com for further assistance.
```
Comment (jouni) [inline articles/configuration/licenses.adoc:125]:
```suggestion
You can contact license@vaadin.com for further assistance.
```

**Review by jouni — COMMENTED** [review]
> I see a lot of information regarding old versions of Vaadin 6,7,8 at https://vaadin.com/pro/licenses. Should we also keep this page in sync with this license guide? Or maybe just link to this guide, so we don't have to repeat ourselves? As of right now, there are three places that discuss licenses.

Thanks, that’s a good observation. We are aware, and we will clean up all of those in the coming weeks, once various implementation pieces start going into production.

File: articles/configuration/licenses.adoc [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
++
+If defined, the `vaadin.proKey` system property will be used instead of the file in the home directory.
+The format of the system property is
++
+[source]
+-Dvaadin.proKey=<login-email>/<key starting with pro->
+
+For CI/CD servers, you need to download a **server license key**.
+Server license keys can only be used for production builds.
+You can download a server license key from https://vaadin.com/myaccount/licenses
```
Comment (tarekoraby) [inline articles/configuration/licenses.adoc:64]:
```suggestion
You can download a server license key (an [filename]#offlineKey# file) from https://vaadin.com/myaccount/licenses.
```

**Review by tarekoraby — APPROVED** [review]

**Pull request merged**